### PR TITLE
Introduce fuzz tests with foundry

### LINKED ISSUE #196: Add fuzz tests to workflow
We need to extend testing to cover unpredictable paths & edge cases of execution.

### PR DESCRIPTION
**Pull request by eduv09**
Fixes https://github.com/skalenetwork/fair-manager/issues/196

### CODE REVIEW

File: test-foundry/handlers/RBTHandler.sol [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    function _remove(NodeId node) private {
+        rbt.remove(node);
+
+        insertedKeys[node] = false;
+        if (NodeId.unwrap(node) < keyList.length - 1) {
+            keyList[NodeId.unwrap(node)] = keyList[keyList.length - 1];
+        }
+        keyList.pop();
```
[the remark this answers is not in the document]
Comment (eduv09) [reply test-foundry/handlers/RBTHandler.sol]: Good catch AI buddy

**Review by DimaStebaev — APPROVED** [review]
